azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -250,8 +280,50 @@ protected void initTelemetry() {
         }
     }
 
+    protected void initAuth() throws AzureAuthFailureException {
+        initializeAuthConfiguration();
+    }
+
     //endregion
 
+    protected void initializeAuthConfiguration() throws AzureAuthFailureException {
+        if (authInitialized) {
+            return;
+        }
+        authInitialized = true;
+        if (!isAuthConfigurationExist()) {
+            return;
+        }
+        if (StringUtils.isNotBlank(auth.getServerId())) {
+            if (this.settings.getServer(auth.getServerId()) != null) {
+                try {
+                    auth = MavenSettingHelper.getAuthConfigurationFromServer(session, settingsDecrypter, auth.getServerId());
+                } catch (MavenDecryptException e) {
+                    throw new AzureAuthFailureException(e.getMessage());
+                }
+            } else {
+                throw new AzureAuthFailureException(String.format("Unable to get server('%s') from settings.xml.", auth.getServerId()));
+            }
+        }
+
+        final List<ConfigurationProblem> problems = auth.validate();
+        if (problems.stream().anyMatch(problem -> problem.getSeverity() == Severity.ERROR)) {
+            throw new AzureAuthFailureException(String.format("Unable to validate auth configuration due to the following errors: %s",
+                    problems.stream().map(ConfigurationProblem::getErrorMessage).collect(Collectors.joining("\n"))));
+        }
```

Comment:
Shall we move the check of server id as well as the validate to ` MavenSettingHelper.getAuthConfigurationFromServer`

Reply:
Should be done later